apidump/net.go

```diff
@@ -54,12 +55,36 @@ func showPermissionErrors(sampleError error) error {
 			printer.Warningf("The agent needs the CAP_NET_RAW capability to capture packets. You are running as an unprivileged (non-root) user.\n")
 			return errors.Errorf("Insufficient permissions, try using \"sudo\" to run as root.")
 		}
+	} else if strings.Contains(sampleError.Error(), "SIOCETHTOOL(ETHTOOL_GET_TS_INFO) ioctl failed: Function not implemented") {
+		// This happens when the binary was built for a different architecture, e.g.
+		// if the user pulled the amd64 Docker image on arm64.
+		arch := architecture.GetCanonicalArch()
+
+		printer.Warningf(
+			"The agent received \"Function not implemented\" when trying to read from your network interfaces. "+
+				"This often indicates that the Akita agent was built for a different architecture than your host architecture. "+
+				"This Akita agent binary was built for %s.\n",
+			arch,
+		)
+
+		if os.Getenv("__X_AKITA_CLI_DOCKER") == "true" {
```

Comment:
```suggestion
		if env.InDocker() {
```

Reply:
Fixed in 0b00a81e70aa176948cf99f1dbef4fba1b00a01e.